src/mbgl/style/expression/compound_expression.cpp

```diff
@@ -675,6 +675,20 @@ const auto& errorCompoundExpression() {
     return signature;
 }
 
+const auto& featureStateCompoundExpression() {
+    static auto signature = detail::makeSignature("feature-state", [](const EvaluationContext& params, const std::string& key) -> Result<Value> {
+        mbgl::Value state = mbgl::Value();
+        if (params.featureState) {
+            auto it = params.featureState->find(key);
+            if (it != params.featureState->end()) {
+                state = mbgl::Value(it->second);
+            }
+        }
+        return toExpressionValue(state);
```

Comment:
is it correct to return null value?

Reply:
Yes - in case no value for the given key was found